File: java/unidad_02/Problema_2_10/src/Problema_2_10.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * El programa, dados como datos tres números cuyos valores pueden ser iguales,
 * determina cuál es el mayor o los mayores.
 */

public class Problema_2_10 {

    public static void encontrarMayor(int numero01, int numero02, int numero03) {

        if (numero01 > numero02) {

            if (numero01 > numero03) {

                System.out.printf("a:%d es el mayor", numero01);

            } else if (numero01 == numero03) {

                System.out.printf("a:%d y c:%d son los mayores", numero01, numero03);

            } else {

                System.out.printf("c:%d es el mayor", numero03);
            }

        } else if (numero01 == numero02) {

            if (numero01 > numero03) {

                System.out.printf("a:%d y b:%d son los mayores", numero01, numero02);

            } else if (numero01 == numero03) {

                System.out.printf("a:%d, b:%d y c:%d son iguales", numero01, numero02, numero03);

            } else {

                System.out.printf("c:%d es el mayor", numero03);
            }

        } else if (numero02 > numero03) {

            System.out.printf("b:%d es el mayor", numero02);

        } else if (numero02 == numero03) {

            System.out.printf("b:%d y c:%d son los mayores", numero02, numero03);

        } else {

            System.out.printf("c:%d es el mayor", numero03);
        }
    }

    public static void main(String[] args) {

        int numero01, numero02, numero03;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("número a: ");
            numero01 = scanner.nextInt();

            System.out.print("número b: ");
            numero02 = scanner.nextInt();

            System.out.print("número c: ");
            numero03 = scanner.nextInt();

        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        encontrarMayor(numero01, numero02, numero03);
    }
}

File: python/unidad_02/problema_2_10.py
'''
El programa, dados como datos tres números cuyos valores pueden ser iguales,
determina cuál es el mayor o los mayores.
'''

def encontrar_mayor(numero_01, numero_02, numero_03):

    if numero_01 > numero_02:

        if numero_01 > numero_03:

            print(f"a:{numero_01} es el mayor")

        elif numero_01 == numero_03:

            print(f"a:{numero_01} y c:{numero_03} son los mayores")

        else: 

            print(f"c:{numero_03} es el mayor")

    elif numero_01 == numero_02:

        if numero_01 > numero_03:

            print(f"a:{numero_01} y b:{numero_02} son los mayores")

        elif numero_01 == numero_03:

            print(f"a:{numero_01}, b:{numero_02} y c:{numero_03} son iguales")
        
        else:

            print(f"c:{numero_03} es el mayor")

    elif numero_02 > numero_03:

        print(f"b:{numero_02} es el mayor")

    elif numero_02 == numero_03:

        print(f"b:{numero_02} y c:{numero_03} son los mayores")

    else:

        print(f"c:{numero_03} es el mayor")

def main():

    try: 

        numero_01 = int(input("número a: "))

        numero_02 = int(input("número b: "))

        numero_03 = int(input("número c: "))

    except ValueError as e:

        print(e)
        return
    
    encontrar_mayor(numero_01, numero_02, numero_03)

if __name__ == '__main__':

    main()